From 9431f815cddb73d79cc5e56923e6c3dcac52d312 Mon Sep 17 00:00:00 2001
From: Akira Ajisaka <aajisaka@apache.org>
Date: Tue, 24 Feb 2015 11:32:11 -0800
Subject: [PATCH 1500/2748] MAPREDUCE-6264. Remove httpclient dependency from
 hadoop-mapreduce-client. Contributed by Brahma
 Reddy Battula.

(cherry picked from commit 3f665aa4d5ac2e737e29ff63b3837c083a49ea80)
(cherry picked from commit 5e686dae44381a0c9a4178ba53bf8befca52fa91)

Change-Id: I80b9482efbbee60489602e57fbb1a6add51f20a2
---
 .../mapreduce/v2/app/webapp/TestAMWebApp.java      |    2 +-
 .../org/apache/hadoop/mapred/JobEndNotifier.java   |   27 ++++++++++----------
 2 files changed, 15 insertions(+), 14 deletions(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/webapp/TestAMWebApp.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/webapp/TestAMWebApp.java
index f955a35..acb31bd 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/webapp/TestAMWebApp.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/webapp/TestAMWebApp.java
@@ -33,7 +33,6 @@
 
 import org.junit.Assert;
 
-import org.apache.commons.httpclient.HttpStatus;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.http.HttpConfig.Policy;
 import org.apache.hadoop.io.IOUtils;
@@ -57,6 +56,7 @@
 import org.apache.hadoop.yarn.webapp.WebApps;
 import org.apache.hadoop.yarn.webapp.test.WebAppTests;
 import org.apache.hadoop.yarn.webapp.util.WebAppUtils;
+import org.apache.http.HttpStatus;
 import org.junit.Test;
 
 import com.google.common.net.HttpHeaders;
diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/JobEndNotifier.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/JobEndNotifier.java
index 74861fe..5f03dc8 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/JobEndNotifier.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/JobEndNotifier.java
@@ -19,17 +19,19 @@
 package org.apache.hadoop.mapred;
 
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.concurrent.Delayed;
 import java.util.concurrent.TimeUnit;
 
-import org.apache.commons.httpclient.HttpClient;
-import org.apache.commons.httpclient.HttpMethod;
-import org.apache.commons.httpclient.URI;
-import org.apache.commons.httpclient.methods.GetMethod;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.classification.InterfaceAudience;
 import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.params.ClientPNames;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.params.CoreConnectionPNames;
 
 @InterfaceAudience.Private
 @InterfaceStability.Unstable
@@ -64,15 +66,14 @@ private static JobEndStatusInfo createNotification(JobConf conf,
   }
 
   private static int httpNotification(String uri, int timeout)
-      throws IOException {
-    URI url = new URI(uri, false);
-    HttpClient httpClient = new HttpClient();
-    httpClient.getParams().setSoTimeout(timeout);
-    httpClient.getParams().setConnectionManagerTimeout(timeout);
-
-    HttpMethod method = new GetMethod(url.getEscapedURI());
-    method.setRequestHeader("Accept", "*/*");
-    return httpClient.executeMethod(method);
+      throws IOException, URISyntaxException {
+    DefaultHttpClient client = new DefaultHttpClient();
+    client.getParams()
+        .setIntParameter(CoreConnectionPNames.SO_TIMEOUT, timeout)
+        .setLongParameter(ClientPNames.CONN_MANAGER_TIMEOUT, (long) timeout);
+    HttpGet httpGet = new HttpGet(new URI(uri));
+    httpGet.setHeader("Accept", "*/*");
+    return client.execute(httpGet).getStatusLine().getStatusCode();
   }
 
   // for use by the LocalJobRunner, without using a thread&queue,
-- 
1.7.9.5

